src/jimp2/wireworld/z8/datamangment/Generator.java

```diff
@@ -1,18 +1,39 @@
 package jimp2.wireworld.z8.datamangment;
 
 
+import jimp2.wireworld.z8.wireworldlogic.CellsContainer;
+import jimp2.wireworld.z8.wireworldlogic.State;
 import jimp2.wireworld.z8.wireworldlogic.World;
 
 import java.awt.*;
 
 public class Generator extends Element {
 
+    private final int width;
+    private final int height;
+
     public Generator(Point position, int width, int height) {
-        // TODO implement here
+        this.position = position;
+        this.width = width;
+        this.height = height;
+        name = "Generator";
+        cellsContainer = new CellsContainer(width, height);
+        for (int row = 0; row < height; row++) {
+            for (int column = 0; column < width; column++) {
+                if (row == 0 || row == height - 1 || column == 0 || column == width - 1)
+                    if (!((row == column) || ((row == 0) && (column == width - 1)) || (row == height - 1 && column == 0)))
```

Comment:
Przenieś proszę te warunki do odpowiednich metod zwracających boolean, które będą miały nazwę sugerującą ich przeznaczenie. Trochę mi zajęło zanim zrozumiałem co się tu dzieje a takie przeniesienie tego do metod to dobra praktyka w myśl podejścia "samokomentującego się kodu" czy jakoś tak :p